ㅁ 신욱이에게 보내기 위해 코드 들여쓰기만 수정을 위한 브런치 커밋

94CJG opened the pull request:
ㅁ 중복확인 눌렀을 때 로컬스토리지 저장 방법 변경
    ㅁ 아이디값 중복확인 기능 추가
    
    ㅁ 문제
    - 중복확인 버튼을 눌렀을 때 초기값 설정을 하는 코드가 없어서 null 또는 빈배열로 들어감 (해결)
    - 중복확인 버튼을 눌렀을 때 유효성 검사 코드가 없어서 1글자 또는 한글 전부 들어
가서 로컬 스토리안에 저장됨(해결)
    - 중복확인 버튼을 눌렀을 때 로컬 스토리지 안에 바로 저장이됨. 회원가입 버튼을 눌렀을 때 유효성 검사 성공시 로컬 스토리지 안에 들어 갈 수 있게 코드 작성(해결)

94CJG committed:
ㅁ 신욱이에게 보내기 위해 코드 들여쓰기만 수정을 위한 브런치 커밋

tlsdnr1135 commented:
가서 로컬 스토리안에 저장됨(해결)
- 중복확인 버튼을 눌렀을 때 로컬 스토리지 안에 바로 저장이됨. 회원가입 버튼을 눌렀을 때 유효성 검사 성공시 로컬 스토리지 안에 들어 갈 수 있게 코드 작성(해결)
-> 이거 기능이 어딨있죠,,? 어제 내가 만든 코드에서 더 추가한거야?

94CJG commented:
오잉.. 푸시한 내용들이 다 안들어갔구나,.
ㅁ 너가 만들어준 코드에서 if문을 추가해서 유효성 검사를 하게 만들었어.
ㅁ 그리고 onValid 함수에 코드 추가 해줬어


```
const handleCheckId = () => {
		//폼에서 입력한 아이디 (중복확인 버튼만 적용)
		const userName = getValues('userName');

		//로컬 스토리지에 값을 가져오기 전에 확인 (중복확인 버튼만 적용)
		const storedUserName = localStorage.getItem('userName');
		const idArray: String[] = storedUserName ? JSON.parse(storedUserName) : [];
		let isExist: boolean = false;

		//폼에서 입력한 아이디와 로컬 스토리지에서 가져온 아이디가 같으면 (중복확인 버튼만 적용)
		idArray && idArray.forEach(
			id => {
				if (id === userName) {
					isExist = true;
					return;
				}
			}
		)

		//아이디가 있으면 (중복확인 버튼만 적용)
		if (isExist) {
			alert("아이디 중복");
		}
		else if (userName.length < 4 || userName.length > 10) {
			alert('아이디는 4글자 이상 10글자 이하 가능 합니다.')
		}
		else if (!/^[a-z0-9]+$/.test(userName)) {
			alert('아이디는 a~z, 0~9 조합이어야 하며, 10글자 이하이어야 합니다. 특수문자는 사용 불가능합니다.');
		}
		else {
			alert("사용 가능한 아이디");
		}
	}

	//유효성 검사 통과(회원가입 버튼 눌렀을 때 적용)
	const onValid = (data: any) => {

		//폼에서 입력한 아이디 값
		const userNameSave = getValues('userName');

		//로컬 스토리지에 값을 가져오기전에 확인
		const storedUserNameGet = localStorage.getItem('userName');
		const idArray: String[] = storedUserNameGet ? JSON.parse(storedUserNameGet) : [];

		//로컬 스토리지에 아이디 값을 저장
		idArray.push(userNameSave);
		localStorage.setItem('userName', JSON.stringify(idArray));

		console.log('로그인 성공', data);
		history.push('/Login');

	}
```

94CJG committed:
ㅁ 회원가입 버튼을 눌렀을 때 로컬 스토리지 아이디 값 저장
- 지피티 이용하여 코드를 작성함
- useState 훅을 사용함.
ㄴ 어떻게 동작하는지 이해 필요함.
- 욱이에게 PR보내기 위해 먼저 커밋함.

tlsdnr1135 commented on src/components/SignUp.tsx:
```diff
@@ -57,9 +59,10 @@ function SignUp() {
 		getValues,
 	} = useForm<IForm>();
+	const history = useHistory();
+	const [isIdValid, setIsIdValid] = useState(true);
```
기본을  false로 주는게 안전할듯.

94CJG committed:
ㅁ 모각 챌린지를 위해 잠시중단.
ㅁ 수정사항
 1. 유효성 검사 else if문을 사용하지 않고, register를 함수로 만들어서
 사용하게 수정해야함.
 2. 또한 유효성 검사 통과하지 못했을 때 중복확인 버튼이 비활성화 기능 구현 해야함
